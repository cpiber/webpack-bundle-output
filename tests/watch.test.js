const { webpack } = require('webpack');
const { toMap, file, webpackOptions } = require('./shared');
const { getFixturePathSync } = require('jest-fixtures');
const { copyFileSync } = require('fs');
const { join } = require('path');

test('watch correctly adds and removes sources', async () => {
  // use a mock to be able to easily exchange callbacks
  const cb = jest.fn();
  const options = {
    entry: getFixturePathSync(__dirname, 'watch'),
  };
  const { out, map } = toMap(options);
  const index = join(options.entry, 'index.js');

  /** @type {import('webpack').Watching} */
  let watcher;
  await new Promise(resolve => {
    // prepare first pass
    cb.mockImplementation(resolve);
    copyFileSync(getFixturePathSync(__dirname, 'watch', 'pass1.js'), index);
    // and start watching
    watcher = webpack(webpackOptions(options, out)).watch({}, cb);
  });
  expect(file(map)).toMatchSnapshot();
  expect(cb).toHaveBeenCalledTimes(1);

  await new Promise(resolve => {
    // prepare second pass
    cb.mockImplementation(resolve);
    copyFileSync(getFixturePathSync(__dirname, 'watch', 'pass2.js'), index);
  });
  expect(file(map)).toMatchSnapshot();
  expect(cb).toHaveBeenCalledTimes(2);

  await new Promise(resolve => {
    // prepare third pass
    cb.mockImplementation(resolve);
    copyFileSync(getFixturePathSync(__dirname, 'watch', 'pass3.js'), index);
  });
  expect(file(map)).toMatchSnapshot();
  expect(cb).toHaveBeenCalledTimes(3);

  await new Promise(resolve => watcher.close(resolve));
});