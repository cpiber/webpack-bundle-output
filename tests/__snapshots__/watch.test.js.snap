// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`watch correctly adds and removes sources 1`] = `
Object {
  "tests/__fixtures__/watch/index.js": Array [
    "tests/out/696bfac4bfa16c7ff712dfff92fb80f2/main.js",
  ],
}
`;

exports[`watch correctly adds and removes sources 2`] = `
Object {
  "tests/__fixtures__/watch/index.js": Array [
    "tests/out/696bfac4bfa16c7ff712dfff92fb80f2/main.js",
  ],
  "tests/__fixtures__/watch/pass1.js": Array [
    "tests/out/696bfac4bfa16c7ff712dfff92fb80f2/main.js",
  ],
}
`;

exports[`watch correctly adds and removes sources 3`] = `
Object {
  "tests/__fixtures__/watch/index.js": Array [
    "tests/out/696bfac4bfa16c7ff712dfff92fb80f2/main.js",
  ],
}
`;
